ci/glci/openstack_image.py

```diff
@@ -0,0 +1,70 @@
+import os
+import time
+
+from glci.model import OpenRC
+
+from openstack import connect
+
+class OpenstackImageUploader():
+  '''OpenstackImageUploader is a client to upload images to Openstack Glance.'''
+
+  def __init__(self, openrc: OpenRC):
+    self.conn = connect(
```

Comment:
nit: I would prefer to not establish the connection in the c'tor, but rather do this when required (a common pattern to do this whilst avoiding repetitive code being to add a method creating the function, and memoising it, using the @functools.lru_cache decorator - there should also be examples in `ci/**.py`)

Reply:
Hmm I'm not sure if the Connection object really holds a persistent connection. I need to check.